src/backend/distributed/commands/database.c

```diff
@@ -208,17 +224,29 @@ PreprocessAlterDatabaseStmt(Node *node, const char *queryString,
 		return NIL;
 	}
 
-	AlterDatabaseStmt *stmt = castNode(AlterDatabaseStmt, node);
-
 	EnsureCoordinator();
 
+	AlterDatabaseStmt *stmt = castNode(AlterDatabaseStmt, node);
+
 	char *sql = DeparseTreeNode((Node *) stmt);
 
 	List *commands = list_make3(DISABLE_DDL_PROPAGATION,
-								(void *) sql,
+								sql,
 								ENABLE_DDL_PROPAGATION);
 
-	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
+	if (isSetTablespaceStatement(stmt))
+	{
+		/* Set tablespace does not work inside a transaction.Therefore, we need to use
```

Comment:
See https://github.com/citusdata/citus/pull/7347.

```suggestion
		/*
		 * Set tablespace does not work inside a transaction.Therefore, we need to use
```